pkg/config/generate.go

```diff
@@ -388,3 +391,43 @@ func Generate(cs *api.OpenShiftManagedCluster, pluginConfig api.PluginConfig) (e
 
 	return
 }
+
+func needsGenerate(cert certificate) bool {
+	crt := *cert.cert
+	// it it is 1st run crt will be empty - generate
+	if crt == nil {
+		return true
+	}
+
+	// if it is n+1 run, and if signingCert signature don't matche - generate
+	if crt.CheckSignatureFrom(cert.signingCert) != nil {
+		return true
+	}
+
+	// if main attributes do not match - regenerate
+	// net.IP does not comply with definition of deep equality.
+	// we use native Equal method within net package
+	if !reflect.DeepEqual(cert.dnsNames, crt.DNSNames) ||
+		!reflect.DeepEqual(cert.extKeyUsage, crt.ExtKeyUsage) ||
+		!reflect.DeepEqual(cert.cn, crt.Subject.CommonName) ||
+		!reflect.DeepEqual(cert.organization, crt.Subject.Organization) ||
+		!ipAddressEqual(cert.ipAddresses, crt.IPAddresses) {
```

Comment:
can this be done better? basically net.IP return not equal all the times when using DeepEqual (similar topic https://stackoverflow.com/questions/42921227/ip-part-of-net-ipnet-does-not-fulfil-reflect-deepequal-but-are-equal) Can this be done better?